### PR TITLE
refactor(Switch): api improvements

### PR DESCRIPTION
**Pull request by elite-benni**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [x] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

The api of the Helm component is very verbose and needs to use many components and directives.
`<label class="flex items-center" hlmLabel>
      <brn-switch class="mr-2" hlm>
        <brn-switch-thumb hlm />
      </brn-switch>
      Airplane mode
    </label>`

Closes #

## What is the new behavior?

The api of the helm part is now simplified. You only need to add the hlm-switch 
`<label class="flex items-center" hlmLabel>
            <hlm-switch class="mr-2" />
            Airplane mode
        </label>`

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

When using the helm component you have to change the api from: 
```
<brn-switch hlm>
   <brn-switch-thumb hlm />
</brn-switch>
```
to:
```
 <hlm-switch />
```

you have to replace the brn component with the hlm-component.

### CODE REVIEW

File: libs/ui/switch/helm/src/lib/hlm-switch.component.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	set class(userCls: ClassValue) {
+		this._userCls.set(userCls);
+	}
+
+	@Output()
+	public changed = new EventEmitter<boolean>();
+
+	handleChange(value: boolean): void {
+		const previousChecked = this._checked();
+		this._checked.set(value);
```
Comment (goetzrobin) [inline libs/ui/switch/helm/src/lib/hlm-switch.component.ts:53]: this feel a little weird. On the one side we use the previousChecked and ignore the value from the change event on the other side we use the value to set the checked value. Maybe we just use one of them? I know they should be in sync anyways but it just feels off.

File: libs/ui/switch/helm/src/lib/hlm-switch.component.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	/** Used to set the aria-labelledby attribute on the underlying brn element. */
+	@Input('aria-labelledby')
+	ariaLabelledby: string | null = null;
+
+	/** Used to set the aria-describedby attribute on the underlying brn element. */
+	@Input('aria-describedby')
+	ariaDescribedby: string | null = null;
+
+	protected _computedClass = computed(() => this._generateClass());
```
Comment (goetzrobin) [inline libs/ui/switch/helm/src/lib/hlm-switch.component.ts:99]: I think we can get rid of the `this._generateClass()` function and just put the hlm directly into the computed.

File: libs/ui/switch/helm/src/lib/hlm-switch.component.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	registerOnChange(fn: any): void {
+		this._onChange = fn;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	registerOnTouched(fn: any): void {
+		this._onTouched = fn;
+	}
+
+	/** Used to set the id on the underlying brn element. */
```
Comment (goetzrobin) [inline libs/ui/switch/helm/src/lib/hlm-switch.component.ts:83]: Let's move all these inputs above the control value accessor stuff

Comment (goetzrobin) [reply libs/ui/switch/helm/src/lib/hlm-switch.component.ts:83]: Including the _computedClass

**Review by goetzrobin — CHANGES_REQUESTED** [review]
I am warming up to this! We're not doing any black magic here, but simply implementing a custom control value accessor the Angular way. I left some comments for a "little clean up". But asides from that I feel like this is a major improvement of the API! Great job!

File: libs/ui/switch/helm/src/lib/hlm-switch.component.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+@Component({
+	selector: 'hlm-switch',
+	imports: [BrnSwitchThumbComponent, BrnSwitchComponent, HlmSwitchThumbDirective],
+	standalone: true,
+	host: {
+		'[attr.class]': 'null',
+		'[attr.id]': 'null',
+		'[attr.aria-label]': 'null',
+		'[attr.aria-labelledby]': 'null',
+		'[attr.aria-describedby]': 'null',
```
Comment (goetzrobin) [inline libs/ui/switch/helm/src/lib/hlm-switch.component.ts:23]: Let's add also add the `'[style.display]': '"contents"'` here. Good German explanation on how this helps here: https://kulturbanause.de/blog/css-display-contents/ :D

File: libs/ui/switch/helm/src/lib/hlm-switch.component.ts (hunk 2 above)
Comment (elite-benni) [reply libs/ui/switch/helm/src/lib/hlm-switch.component.ts:99]: My usual plan was to get rid of _generateClass for all components, once we are on 17.1 and we do the signal inputs refactoring. ;)
but i can do it for this now, as you already started to remove the extra function in other components I don't see a problem.

File: libs/ui/switch/helm/src/lib/hlm-switch.component.ts (hunk 1 above)
Comment (elite-benni) [reply libs/ui/switch/helm/src/lib/hlm-switch.component.ts:53]: Yes this is weird, must have been something I left when trying so many different things. 
Need to clean up here.

**Comment by elite-benni** [comment]
Additional "breaking change" is that we now use display contents, wich caused the Switch to be displayed without gap in the card example, the gap was added manually in the example.

File: apps/app/src/app/pages/(components)/components/(card)/card--notifications.example.ts [hunk 5]
```diff
@@ -91,95 +85,89 @@ export class CardNotificationsComponent {
...
+					<hlm-icon size="lg" name="radixBell" />
+					<div class="flex-1 space-y-1">
+						<p class="text-sm font-medium leading-none">Push Notifications</p>
+						<p class="text-muted-foreground text-sm">Send notifications to device.</p>
+					</div>
+					<hlm-switch aria-label="Enable notifications" />
+				</div>
+				<div>
+					<div
+						*ngFor="let notification of notifications"
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(components)/components/(card)/card--notifications.example.ts:137]: Not a blocker but we can move this to use the new control flow

File: libs/ui/switch/brain/src/lib/brn-switch-ng-model.spec.ts [hunk 6]
```diff
@@ -51,9 +51,11 @@ describe('BrnSwitchComponentNgModelIntegration', () => {
 		expect(container.fixture.componentInstance.airplaneMode).toBe(true);
 	});
 
+	// TODO - why does this test not fail? it should be setup(false, true) but setup(false, false) still passes
 	it('should do nothing when disabled', async () => {
```
Comment (goetzrobin) [inline libs/ui/switch/brain/src/lib/brn-switch-ng-model.spec.ts:55]: I can check these out after we merge. Seems like some CD issues... As long as the e2e tests pass I am fine with it

File: libs/ui/switch/helm/jest.config.ts [hunk 7]
```diff
@@ -14,6 +14,7 @@ export default {
 		],
 	},
 	transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$)'],
+	testPathIgnorePatterns: ['<rootDir>/src/lib/hlm-switch.ng-model.component.spec.ts'],
```
Comment (goetzrobin) [inline libs/ui/switch/helm/jest.config.ts:17]: Can we use something that is more obvious in the test file, e.g. `test.skip`? I could see this be a source of confusion on why tests are not being executed. Again, not a blocker!

**Review by goetzrobin — APPROVED** [review]
Left some minor comments! But happy to merge this as is! End result of this API change looks sexy 😍 Great work @elite-benni!

File: libs/ui/switch/helm/jest.config.ts (hunk 7 above)
Comment (elite-benni) [reply libs/ui/switch/helm/jest.config.ts:17]: yes I renamed it to ignore since it is added to IgnorePatterns. 
also changed this pattern in brn.

**Pull request merged**